room/room-compiler-processing/src/main/java/androidx/room/compiler/processing/XValueArgument.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.room.compiler.processing
+
+/**
+ * This wraps information about an argument in an annotation.
+ */
+interface XValueArgument {
```

Comment:
I think `XAnnotationValue` might be a more fitting name, mimicking https://docs.oracle.com/javase/8/docs/api/javax/lang/model/element/AnnotationValue.html or maybe `XAnnotationProperty` which might describe bettere that is both a method and some default value (a getter?).

Reply:
@danysantiago `ValueArgument` is the term that KSP uses in it's API - would you rather follow KSP or Java API naming?

Reply:
So far we've have been following Java API when available, TypeElement vs ClassDeclaration, ExecutableElement vs FunctionDeclaration so I was thinking to have the same here, specially when this is only used for annotation values where as KSP's ValueArgument could be used to describe arguments in function calls (doesn't right now, but might...)

Reply:
ok, I'll use `XAnnotationValue`

Reply:
Thank you!